ui/lib/core/addon/components/bar-chart.js

```diff
@@ -0,0 +1,396 @@
+      counts: {
+        distinct_entities: 547,
+        non_entity_tokens: 337,
+        clients: 884,
+      },
+    },
+    {
+      namespace_id: '1oihz',
+      namespace_path: 'someOtherNamespaceawgagawegawgawgawgaweg',
+      counts: {
+        distinct_entities: 807,
+        non_entity_tokens: 234,
+        clients: 1041,
+      },
+    },
+  ];
+
+  get labelKey() {
+    return this.args.labelKey || 'label';
+  }
+
+  get getButtonText() {
+    return this.args.getButtonText || 'Export data';
+  }
+
+  get mapLegend() {
+    return this.args.mapLegend || null;
+  }
+
+  get chartKeys() {
+    return this.args.chartKeys || null;
+  }
+
+  // TODO: take in arguments passed to component
+  flattenedData() {
+    return this.dataset.map(d => {
+      return {
+        label: d['namespace_path'],
+        non_entity_tokens: d['counts']['non_entity_tokens'],
+        distinct_entities: d['counts']['distinct_entities'],
+        total: d['counts']['clients'],
+      };
+    });
+  }
+
+  // TODO: separate into function for specifically creating tooltip text
+  totalCount = this.dataset.reduce((prevValue, currValue) => prevValue + currValue.counts.clients, 0);
+
+  @action
+  renderBarChart(element) {
+    let totalCount = this.totalCount;
+    let handleClick = this.args.onClick;
+    let labelKey = this.labelKey;
+    let dataset = this.flattenedData();
+    let elementId = guidFor(element);
+
+    let stackFunction = stack().keys(this.mapLegend.map(l => l.key));
+    // creates an array of data for each map legend key
+    // each array contains coordinates for each data bar
+    let stackedData = stackFunction(this.flattenedData());
+
+    // creates and appends tooltip
+    let container = select('.bar-chart-container');
+    container
+      .append('div')
+      .attr('class', 'chart-tooltip')
+      .attr('style', 'position: fixed; opacity: 0;')
+      .style('color', 'white')
+      .style('background', `${TOOLTIP_BACKGROUND}`)
+      .style('font-size', '.929rem')
+      .style('padding', '10px')
+      .style('border-radius', '4px');
+
+    let xScale = scaleLinear()
+      .domain([0, max(dataset.map(d => d.total))])
+      .range([0, 75]); // 25% reserved for margins
+
+    let yScale = scaleBand()
+      .domain(dataset.map(d => d[labelKey]))
+      .range([0, dataset.length * LINE_HEIGHT])
+      .paddingInner(0.765); // percent of the total width to reserve for white space between bars
+
+    let chartSvg = select(element);
+    chartSvg.attr('viewBox', `0 0 710 ${(dataset.length + 1) * LINE_HEIGHT}`);
+    chartSvg.attr('id', elementId);
+
+    // creates group for each array of stackedData
+    let groups = chartSvg
+      .selectAll('g')
+      .data(stackedData)
+      .enter()
+      .append('g')
+      // shifts chart to accommodate y-axis legend
+      .attr('transform', `translate(${CHART_MARGIN.left}, ${CHART_MARGIN.top})`)
+      .style('fill', (d, i) => BAR_COLOR_DEFAULT[i]);
```

Comment:
For colors, can you create a class name and set that, moving the style declarations into CSS?